airbyte-webapp/src/views/Connection/ConnectionForm/components/ScheduleField.tsx

```diff
@@ -0,0 +1,182 @@
+import { Field, FieldInputProps, FieldProps, FormikProps } from "formik";
+import { ChangeEvent, useMemo } from "react";
+import { useIntl } from "react-intl";
+
+import { ControlLabels, DropDown, DropDownRow, Input } from "components";
+
+import { ConnectionScheduleData, ConnectionScheduleType } from "core/request/AirbyteClient";
+
+import availableCronTimeZones from "../../../../config/availableCronTimeZones.json";
+import { ConnectionFormMode } from "../ConnectionForm";
+import { FormikConnectionFormValues, useFrequencyDropdownData } from "../formConfig";
+import styles from "./ScheduleField.module.scss";
+
+interface ScheduleFieldProps {
+  scheduleData: ConnectionScheduleData | undefined;
+  mode: ConnectionFormMode;
+  onDropDownSelect?: (item: DropDownRow.IDataItem) => void;
+}
+
+const CRON_DEFAULT_VALUE = {
+  cronTimeZone: "UTC",
+  // Fire at 12:00 PM (noon) every day
+  cronExpression: "0 0 12 * * ?",
+};
+
+const ScheduleField: React.FC<ScheduleFieldProps> = ({ scheduleData, mode, onDropDownSelect }) => {
+  const { formatMessage } = useIntl();
+  const frequencies = useFrequencyDropdownData(scheduleData);
+
+  const onScheduleChange = (item: DropDownRow.IDataItem, form: FormikProps<FormikConnectionFormValues>) => {
+    onDropDownSelect?.(item);
+
+    let scheduleData: ConnectionScheduleData;
+    const isManual = item.value === ConnectionScheduleType.manual;
+    const isCron = item.value === ConnectionScheduleType.cron;
+
+    // Set scheduleType for yup validation
+    const scheduleType = isManual || isCron ? (item.value as ConnectionScheduleType) : ConnectionScheduleType.basic;
+
+    // Set scheduleData.basicSchedule
+    if (isManual || isCron) {
+      scheduleData = {
+        basicSchedule: undefined,
+        cron: isCron ? CRON_DEFAULT_VALUE : undefined,
+      };
+    } else {
+      scheduleData = {
+        basicSchedule: item.value,
+      };
+    }
+
+    form.setValues(
+      {
+        ...form.values,
+        scheduleType,
+        scheduleData,
+      },
+      true
+    );
+  };
+
+  const getBasicScheduleValue = (value: ConnectionScheduleData, form: FormikProps<FormikConnectionFormValues>) => {
+    const scheduleType = form.values.scheduleType;
+
+    if (scheduleType === ConnectionScheduleType.basic) {
+      return value.basicSchedule;
+    }
+
+    return formatMessage({
+      id: `frequency.${scheduleType}`,
+    }).toLowerCase();
+  };
+
+  const getZoneValue = (currentSelectedZone: string | undefined): string => {
+    if (!currentSelectedZone) {
+      return "UTC";
+    }
+
+    return currentSelectedZone;
+  };
```

Comment:
Nit: Could default this value to "UTC".
```suggestion
  const getZoneValue = (currentSelectedZone = "UTC") => currentSelectedZone;
```